src/Temporalio/Worker/ActivityWorker.cs

```diff
@@ -366,13 +366,14 @@ private void StartActivity(Bridge.Api.ActivityTask.ActivityTask tsk)
                     Headers: tsk.Start.HeaderFields)).ConfigureAwait(false);
 
                 completion.Result.Completed = new();
-                // As a special case, ValueTuple is considered "void"
-                if (result?.GetType() != typeof(ValueTuple))
+                // As a special case, ValueTuple is considered "void" which is null
+                if (result?.GetType() == typeof(ValueTuple))
                 {
-                    completion.Result.Completed.Result =
-                        await worker.Client.Options.DataConverter.ToPayloadAsync(
-                            result).ConfigureAwait(false);
+                    result = null;
                 }
+                completion.Result.Completed.Result =
```

Comment:
Should we add an explicit test to cover this? I guess the semaphore tests cover it as well?

Reply:
Yes, the semaphore test hit this because it was our first `Task`-no-result activity (as opposed to just `void`). I can add an explicit test.

Reply:
Hrmm, so `ExecuteActivityAsync_SimpleAsyncVoidMethod_Succeeds` tested this and succeeds in CI but I don't know why, digging...

Reply:
Ok, there are different routes to creating an activity, and that other test didn't test this route. This semaphore test should be sufficient.